### DISCUSSION TITLE
What will happen if I delete the tracked files from .git database?

### DISCUSSION THREAD

**Discussion by Cactusinhand** [git-lfs/git-lfs#4715] [Q&A] [answered]
Hi, all
I am wonder what will happen if I use `git lfs migrarte import` to track existing large files in my Git repo, and then I delete these big files totally from `.git/objects` database, of course, the commit history will also rewritten too, will those files still be tracked by Git-LFS? and on the server side, Git-LFS Server would still store those file contents?

**Comment by chrisd8088** [comment, marked as answer]
There are a few questions here, I think.  I believe what's being asked is, in essence, what is the lifecycle of large object files when they are managed by Git LFS?

Here's a local example where `git push` is never run, so the data is only ever local.
```
$ git init test
$ cd test
$ git lfs track '*.bin'
$ git add .gitattributes
$ git commit -m attribs

$ echo foo >foo.bin
$ git add foo.bin
$ git commit -m foo
$ git lfs ls-files
b5bb9d8014 * foo.bin

$ git reset --hard HEAD^    # drop previous commit
HEAD is now at 711bb47 attribs

$ git reflog    # however, Git still retains dropped commit
711bb47 (HEAD -> master) HEAD@{0}: reset: moving to HEAD^
170b8a3 HEAD@{1}: commit: foo
711bb47 (HEAD -> master) HEAD@{2}: commit (initial): attribs
$ git cat-file -p 170b8a3
tree 9ea876739ff0b307b50571ac05d137d9b9258e71
parent 711bb473937253c49e6abbdea58d95f0bb405ad8
...
$ git cat-file -p 9ea8767
100644 blob 4edd5acb13dba6a9f44a206fa1b6a1789fbcc50a	.gitattributes
100644 blob ad483a923bcf4e7ef83182817802d740ee4d2ec6	foo.bin
$ git cat-file -p ad483a9
version https://git-lfs.github.com/spec/v1
oid sha256:b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c
size 4
$ cat .git/lfs/objects/b5/bb/b5bb9d8*    # Git LFS object file still exists
foo

$ git reflog expire --expire=now --all    # prune reflog and Git objects
$ git gc --prune=now
$ git cat-file -p ad483a9
fatal: Not a valid object name ad483a9
$ cat .git/lfs/objects/b5/bb/b5bb9d8*    # Git LFS object file still exists
foo

$ git lfs prune    # prune Git LFS object file
cat .git/lfs/objects/b5/bb/b5bb9d8*
cat: .git/lfs/objects/b5/bb/b5bb9d8*: No such file or directory
```

Now let's consider what might happen if you pushed to your remote after committing the file using Git LFS.  (Note that Git LFS won't automatically push copies of your large files to your remote; you need to run `git push` to do that.)  So if you run `git push origin main` after the `git commit -m foo` then your remote will have a copy of the Git LFS object file in its LFS storage.

If you then followed the remaining steps and purged your local repository, and then did a `git push --force` to your remote, the remote will no longer have a copy of the Git object `ad483a923bcf4e7ef83182817802d740ee4d2ec6` that referenced the Git LFS object `b5bb9d8014a0f9b1d61e21e796d78dccdf1352f23cd32812f4850b878ae4944c`.  But it may well retain a copy of the Git LFS object indefinitely; so far as I know, there's nothing in the Git LFS protocol that requires a Git LFS server to purge objects on any particular schedule.  So different Git LFS hosts may behave differently, and some may not remove the Git LFS objects until you delete the whole repository.

**Reply by Cactusinhand** [reply]
thanks a lot 🎉
